### PR TITLE
limit the available options for voluntary-exit.

### PR DESCRIPTION
**Pull request by rolfyone**
Signed-off-by: Paul Harris <paul.harris@consensys.net>

## Documentation

- [X] I thought about documentation and added the `documentation` label to this PR if updates are required.

### CODE REVIEW

File: teku/src/main/java/tech/pegasys/teku/cli/options/ValidatorKeysOptions.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      names = {"--validator-keys"},
+      paramLabel = "<KEY_DIR>:<PASS_DIR> | <KEY_FILE>:<PASS_FILE>",
+      description =
+          "<KEY_DIR>:<PASS_DIR> will find <KEY_DIR>/**.json, and expect to find <PASS_DIR>/**.txt.\n"
+              + "<KEY_FILE>:<PASS_FILE> will expect that the file <KEY_FILE> exists, "
+              + "and the file containing the password for it is <PASS_FILE>.\n"
+              + "The path separator is operating system dependent, and should be ';' in windows rather than ':'.",
+      split = ",",
+      arity = "1..*")
+  private List<String> validatorKeys = new ArrayList<>();
```
Comment (ajsutton) [inline teku/src/main/java/tech/pegasys/teku/cli/options/ValidatorKeysOptions.java:41]: This isn't enough.  We also need the options to specify external signers:
* `validators-external-signer-public-keys`
* `--validators-external-signer-url`
* `--validators-external-signer-timeout`

File: teku/src/main/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommand.java [hunk 2]
```diff
@@ -179,10 +164,10 @@ private void initialise() {
     }
     genesisRoot = maybeRoot.get();
 
-    // get the validator
-    final Path slashProtectionPath = getSlashingProtectionPath(dataOptions);
+    // slashing protection won't be looked at, but it is required to create a validator loader.
+    // No files will be created in the path
     final SlashingProtector slashingProtector =
-        new SlashingProtector(new SyncDataAccessor(), slashProtectionPath);
+        new SlashingProtector(new SyncDataAccessor(), Path.of("."));
```
Comment (ajsutton) [inline teku/src/main/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommand.java]: Probably worth making `SlashingProtector` an interface so we can just supply a `FORBID_ALL` option here (validator exit shouldn't ask to sign any attestations or blocks so if it does, slashing protector should reject the request)

File: teku/src/main/java/tech/pegasys/teku/cli/options/ValidatorKeysOptions.java (hunk 1 above)
Comment (ajsutton) [reply teku/src/main/java/tech/pegasys/teku/cli/options/ValidatorKeysOptions.java:41]: May as well move the old `validators-key-files` and `validators-key-password-files` options into here too just for consistency.  They're hidden and will be removed soon but its tidier to keep all the key loading options in one place.

File: teku/src/main/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommand.java (hunk 2 above)
Comment (rolfyone) [reply teku/src/main/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommand.java]: hopefully i encapsulated pretty much what you're saying here.

File: teku/src/main/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommand.java [hunk 3]
```diff
@@ -98,10 +84,33 @@
...
     initialise();
+    confirmExits();
     getValidatorIndices(blsPublicKeyValidatorMap).forEach(this::submitExitForValidator);
   }
 
+  private void confirmExits() {
+    SUB_COMMAND_LOG.display("Exits are going to be generated for validators: ");
+    SUB_COMMAND_LOG.display(getValidatorAbbreviatedKeys());
+    SUB_COMMAND_LOG.display(
+        "\nThese validators won't be able to be re-activated, and withdrawals aren't likely to be possible until Phase 2 of eth2 Mainnet.");
```
Comment (ajsutton) [inline teku/src/main/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommand.java]:
```suggestion
        "\nThese validators won't be able to be re-activated, and withdrawals are unlikely to be possible until Phase 2 of eth2 Mainnet.");
```

Comment (ajsutton) [reply teku/src/main/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommand.java]: Also, can't hard code `\n` - should separate calls to display for each line.

**Review by ajsutton — APPROVED** [review]
LGTM.

File: teku/src/main/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommand.java (hunk 3 above)
Comment (rolfyone) [reply teku/src/main/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommand.java]: that was for an extra line, i could put in a blank display call?

Comment (ajsutton) [reply teku/src/main/java/tech/pegasys/teku/cli/subcommand/VoluntaryExitCommand.java]: Yeah a blank call would work.  `\n` isn't the line separator on all systems.

**Pull request merged**